tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistry.java

```diff
@@ -108,4 +108,12 @@
      * Removes a TaggedMetricsSet added via addMetrics from this metrics set.
      */
     Optional<TaggedMetricSet> removeMetrics(String safeTagName, String safeTagValue);
+
+    /**
+     * Removes a TaggedMetricsSet added via addMetrics from this metrics set, if currently registered to this metric
+     * set.
+     *
+     * @return true if value was removed
+     */
+    boolean removeMetrics(String safeTagName, String safeTagValue, TaggedMetricSet metrics);
```

Comment:
can we document the assumption around equality comparison check for `TaggedMetricSet` a la https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#remove-java.lang.Object-java.lang.Object-